src/Evolution/Systems/Cce/NpValues.cpp

```diff
@@ -0,0 +1,54 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+#include "Evolution/Systems/Cce/NpValues.hpp"
+
+#include "DataStructures/ComplexDataVector.hpp"
+#include "DataStructures/SpinWeighted.hpp"
+#include "DataStructures/Tensor/Tensor.hpp"
+#include "Evolution/Systems/Cce/Tags.hpp"
+#include "Utilities/Gsl.hpp"
+#include "Utilities/TMPL.hpp"
+
+namespace Cce {
+
+void weyl_psi0_impl(
+    const gsl::not_null<SpinWeighted<ComplexDataVector, 2>*> psi_0,
+    const SpinWeighted<ComplexDataVector, 2>& bondi_j,
+    const SpinWeighted<ComplexDataVector, 2>& dy_j,
+    const SpinWeighted<ComplexDataVector, 2>& dy_dy_j,
+    const SpinWeighted<ComplexDataVector, 0>& bondi_k,
+    const SpinWeighted<ComplexDataVector, 0>& bondi_r,
+    const SpinWeighted<ComplexDataVector, 0>& one_minus_y) noexcept {
+  // note: intential use of expression template instead of allocation
```

Comment:
`dy_beta` appears below twice, so don't you want an allocation?

Reply:
I think I can factor the expression to only make use of it once instead.